api/models/templates/service/s3.tmpl

```diff
@@ -0,0 +1,119 @@
+{{ define "service" }}
+  {
+    "AWSTemplateFormatVersion" : "2010-09-09",
+    "Conditions": {
+      "Queue": { "Fn::Not": [ { "Fn::Equals": [ { "Ref": "Queue" }, "" ] } ] }
+    },
+    "Parameters": {
+      "Queue": {
```

Comment:
Ok, I'm going to to a little timebox on SNS while this stuff is still fresh on my mind.